### DISCUSSION TITLE
[Resolved] Source Code Events: What is a change?

### DISCUSSION THREAD

**Discussion by erkist** [cdfoundation/sig-events#34] [General]
As per my action from an earlier Vocabulary Workgroup meeting, we should try to settle what we mean by a "change".

The background of this is discussion held in PR https://github.com/cdfoundation/sig-events/pull/7, it may be relevant to look into the PR as well.

@afrittoli wrote:
> A change is an abstraction created by tools like gerrit, github and gitlab on top of git.
> We need to have change specific events, but I wonder if we should have events for the underlying repo too.
> For instance when a change is closed, you could from zero to many commits added to the repo.
> Do we want separate events for that, or do we want to embed that information in the change event?
> You can have commits added to a repo without a change too, as most of this systems allow direct interactions with the underlying source code version control system.
> 
> The data model can easily explode in this space (see all github events for instance), so I think it's fine if we can it simple in the beginning, but I think it's important to make sure we have an event when something changes in the repository.

So, based on Andrea's statement above, what do we see as a change, and what events should the spec have that are connected to it?

**Comment by m-linner-ericsson** [comment]
In https://github.com/cdfoundation/sig-events/blob/main/vocabulary-draft/source-code-version-control.md we have defined a couple of events related to a change borrowing the vocabulary from Gerrit as in GitHub it is called a Pull Request. Would be more agnostic to call it a proposal instead?

**Reply by erkist** [reply]
I like it! Either "proposal" or "change proposal" seem fine to me, with a slight preference for the latter (to include the word "change").

This way, a proposal / change proposal, if approved, would result in a change, which is terminology that makes sense to me

**Reply by m-linner-ericsson** [reply]
How do we go from here? Create a PR (or using our own term "change proposal")?

**Reply by erkist** [reply]
PR created https://github.com/cdfoundation/sig-events/pull/85
